effect_decision.py

```diff
@@ -139,14 +160,163 @@ def get_effect_list(self, area_data, audio_beats, au_instance, framerate, group_
                 zoom_scale_list.append(scale)
 
                 effect_list.append(effect_dict)
-            
+
             pre_end_to = end_to
             pre_start_from = start_from
             pre_key_frame_idx = key_frame_idx
-            #ToDo: debug start
             if self.debug:
                 print("group {}:".format(str(i)),left,right,start_from, key_frame_idx, end_to)
-            #ToDo: debug end
-        # print(effect_list)
+                print(effect_list)
+        self.effect_desc_list = effect_list
+        if self.debug is True:
+            print(self.effect_desc_list)
+        return effect_list, start_list, key_list, end_list, zoom_scale_list, group_list
+
+    def make_effect_point_list_from_desc_list(self):
+        default_scale = self.default_scale
+        default_loc_x = self.default_loc_x
+        default_loc_y = self.default_loc_y
+
+        effect_point_list = []
+        # Set properties for first two frames
+        effect_point_list.extend([
+            {'frame': 0, 'scale_x': 1.0, 'scale_y': 1.0, 'location_x': 0.0, 'location_y': 0.0},
+            {'frame': 1, 'scale_x': default_scale, 'scale_y': default_scale, 'location_x': default_loc_x,
+             'location_y': default_loc_y},
+        ])
+
+        # Set properties for start_from, keyframe, and end_to that are described in self.effect_desc_list
+        last_key_scale = default_scale
+        for effect_desc in self.effect_desc_list:
+            # effect_desc={
+            # 'frame': key_frame_idx,
+            # 'effect': [{'type':'zoom', 'scale':scale}],
+            # 'start_from': start_from,
+            # 'end_to': end_to}
+
+            # 1. According to effect_desc, get the scale at start_from, keyframe, and end_to.
```

Comment:
If we parse the scale and other info directly rather than pass the effect_desc, maybe we can reduce code from line 197 to 207?